main.tf

```diff
@@ -25,6 +25,22 @@ locals {
     id             = coalesce(var.launch_template_id, aws_launch_template.default[0].id)
     latest_version = coalesce(var.launch_template_version, aws_launch_template.default[0].latest_version)
   }
+
+  eks_node_group_resources = [for resource in aws_eks_node_group.default.*.resources :
```

Comment:
the sg id is needed as an output for users that might want to add rules to it outside the module. i was appending to resources to match prior behavior. happy to put it as its own output. 
it's important to have as an output for users that might want to add additional rules outside the module.  if you're ok breaking backward compatibility, this can all be updated and users can just provide an sg id to the module to attach to the launch template and bypass most of the logic in this PR